### ISSUE TITLE
No parse error on `(.|.`

### ISSUE THREAD

**Issue by Mortal** [rust-lang-nursery/regex#239]
In rust-regex 0.1.71 (2af46a44a), I expect `Regex::new(r'(.|.')` to return a parse error, but it does not.

Steps to reproduce:

Add the following line to tests/noparse.rs:

```
noparse!(fail_unclosed_paren, r"(.|.");
```

Run `cargo test noparse::fail_unclosed_paren`. I expect the test to succeed, but instead, the test doesn't fail.

Instead, `Regex::new(r"(.|.")` succeeds and returns a compiled regex object that matches nothing. (I found this the hard way while writing a lexer and struggling to escape my special characters properly...)

**Comment by BurntSushi** [comment]
Nice find! `regex-debug` thinks the AST is:

```
$ regex-debug ast '(.|.'
Alternate(
    [
        AnyCharNoNL,
        AnyCharNoNL
    ]
)
```

**Comment by BurntSushi** [comment]
Fixed in #240.

The fix is on crates.io in `regex-syntax 0.3.3`.
